docs/developer/products.mdx

```diff
@@ -509,14 +513,94 @@ Response:
 }
 ```
 
-- `"quantityAvailable": 50`: customers only see maximum of 50 pcs available to buy
-- `stocks` contain full list of stock quantities in every of the warehouses
+- The `quantityAvailable` is a sum of stock quantities from all available stocks.
+  In this case, the sum is bigger than the `MAX_CHECKOUT_LINE_QUANTITY` so the maximum
+  value of 50 products is returned.
+- `stocks` contain a full list of stock quantities in each warehouse.
+
+:::note
+When the product has a track inventory turned off, the `quantityAvailable` will be always
+equal to the `MAX_CHECKOUT_LINE_QUANTITY` value.
+:::
+
+### Getting the available quantity for the particular address
+
+The quantity available can be fetched for the given address. The returned quantity
+is the sum of the quantity from all stocks that are available in the shipping zone that
+supports provided address. When the address is not provided the highest quantity
+from available shipping zones is returned.
+
+To get the `quantityAvailable` for a particular address use the `address` parameter:
+Request:
+
+```graphql {3-5}
+query {
+    productVariant(id: "UHJvZHVjdFZhcmlhbnQ6MjAz", channel: "default-channel") {
+    quantityAvailable(
+      address: {"country": "US"}
+    )
+    stocks {
+      warehouse {
+        slug
+      }
+      quantity
+      quantityAllocated
+    }
+  }
+}
+```
+
+Response:
+
+```json {4,10-11,17-18}
+{
+  "data": {
+    "productVariant": {
+      "quantityAvailable": 20,
+      "stocks": [
+        {
+          "warehouse": {
+            "slug": "europe"
+          },
+          "quantity": 30,
+          "quantityAllocated": 0
+        },
+        {
+          "warehouse": {
+            "slug": "americas"
+          },
+          "quantity": 20,
+          "quantityAllocated": 0
+        }
+      ]
+    }
+  }
+}
+```
+
+In the response, we get the available quantity of 20 as the `americas` is the only
+warehouse available in the `US`.
+
+In the case when the address is not provided, the quantity for the shipping zone with
+the highest available quantity is returned.
+
+### Handling collection point warehouses in the quantity calculations
+
+The collection point warehouses are treated as a magic single-warehouse shipping zone.
```

Comment:
If we say that something is "magic" I feel like we don't explain it properly, things should not be magic to our users :) 
```suggestion
The collection point warehouses are treated as a single-warehouse shipping zone.
```